src/app/g3w-ol/controls/basequerypolygoncontrol.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * @file
+ * @since v3.8
+ */
+import { SPATIALMETHODS, VM } from 'g3w-ol/constants';
+
+const InteractionControl = require('g3w-ol/controls/interactioncontrol');
+const { merge }          = require('core/utils/ol');
+const { Geometry }       = require('core/utils/geo');
+
+const VALIDGEOMETRIES    = Geometry.getAllPolygonGeometryTypes();
+
+const BaseQueryPolygonControl = function(options = {
+    spatialMethod=SPATIALMETHODS[0],
+    name,
+    tipLabel,
+    label,
+    onSelectlayer,
+    enabled=true,
+    interactionClass
+  } = {}) {
+
+  /**
+   * @type {unknown[]}
+   *
+   * @since 3.8.0
+   */
+  this.layers = options.layers || [];
+
+  /**
+   * @type {unknown[]}
+   *
+   * @since 3.8.0
+   */
+  this.externalLayers = [];
+
+  /**
+   * @type {unknown[]}
+   *
+   * @since 3.8.0
+   */
+  this.unwatches = [];
```

Comment:
The `unwatches` variable appears to be unusued within this file: `src/app/g3w-ol/controls/basequerypolygoncontrol.js`

Reply:
@Raruto remove